test/runtests.jl

```diff
@@ -32,19 +32,14 @@ H_GT = [6 10 8 2 0 1 2 10;
      4 9 10 7 7 0 0 0
     ]
 
-@testset "test top wrapper" begin
+@testset "test top wrapper" begin        
     W = W_GT[:, 3:4]
     H = H_GT[3:4, :]
     X = W*H
-    standard_nmf = nnmf(float(X), 1; init=:nndsvd, tol=1e-12, initdata = svd(float(X)))
-    result_renmf = nmfmerge(float(X), 2=>1; alg = :cd, maxiter = 10^5, tol_final=1e-12, tol_intermediate = 1e-12);
-    W_standard, H_standard = standard_nmf.W, standard_nmf.H
+    result_renmf = nmfmerge(float(X), 3=>2; alg = :cd, maxiter = 10^5, tol_final=1e-12, tol_intermediate = 1e-12);
     W_renmf, H_renmf = result_renmf.W, result_renmf.H
-    W_standard, H_standard = colnormalize(W_standard, H_standard)
-    W_renmf, H_renmf = colnormalize(W_renmf, H_renmf)
-    @test sum(abs2, W_standard - W_renmf) <= 1e-12
-    @test sum(abs2, H_standard - H_renmf) <= 1e-12
-
+    @test size(W_renmf, 2) == 2
+    @test size(H_renmf, 1) == 2
 
```

Comment:
Here I tested rank_1 standard NMF and nmfmerge 2=>1 give equivalent answer, but gsvd on rank_1 NMF solution duplicate the component. Then, over-complete NMF will have a zero component and normalization give that zero component away. Since we only need to test if the function works or not, I test nmfmerge 3=>2 case and we don't know the exact answer of rank 2 NMF. I think just test the final W components number is enough.

Reply:
Can you test that the objective value is below some threshold? At least that might catch a regression.

Reply:
I added a test. Fortunately, the rank-2 NMF gives a good factorization here. The W and H are fitted well, but the order of the columns changes after NMF, I don't want to copy the whole column match code here. So, I only test the reconstruction matrix.